py-client/client_api.py

```diff
@@ -608,12 +650,16 @@ def http_multipart(method, server_url, selector, fields, files, multipart_respon
             else:
                 return response.status, response.read(), None
 
-        return response.status, response.read()
+        logger.debug('Reading status/content from simple response!')
+        return response.status, response.read(), None
 
     @staticmethod
     def save_result(files, result_file):
         logger = logging.getLogger(__name__)
 
+        if result_file is None:
+            return
```

Comment:
do we want this to be silent return here?
or we want to do checks before calling this function?
Or we can add a docstring describe this behavior

Reply:
it depends on api.. if they are expecting any output file (image)... it's more of internal class/util.. didn't want users to use this...

Reply:
got it, make sense